### PR TITLE
refactor(useForm): Replace `Proxy` with property getters.

### PR DESCRIPTION
**Pull request by avocadowastaken**
Currently, `useForm` has different behaviours if `Proxy` is supported or not, these changes will make it similar for both cases.

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -94,14 +94,13 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
     typeof document !== UNDEFINED &&
     !isWindowUndefined &&
     !isUndefined(window.HTMLElement);
-  const isProxyEnabled = isWeb && 'Proxy' in window;
```
Comment (bluebill1049) [inline src/useForm.ts]: you will need this for react native :)

**Comment by bluebill1049** [comment]
thanks for the PR 👍 but i think we do need feature detect for proxy. (react-native)

**Comment by umidbekkarimov** [comment]
@bluebill1049 Oh, it had never occurred to me. Is it because React Native will always have controlled components?  At least we should add some comments regarding that.

The reason why I wanted to replace Proxy is that we lint our built file with `eslint-plugin-compat` to ensure compatibility with some old browsers and there are no good Proxy polyfill that covers all edge cases.

File: src/useForm.ts [hunk 2]
```diff
@@ -1076,17 +1100,6 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
...
-            if (prop in obj) {
-              readFormState.current[prop] = true;
-              return obj[prop];
-            }
-
-            return {};
-          },
-        })
-      : formState,
+    formState,
```
Comment (bluebill1049) [inline src/useForm.ts]: To be honest i think let's stay for what we having right now. This is introduce quite bit extra code for this lib (size perspective), but thanks for the PR tho.

**Comment by bluebill1049** [comment]
> @bluebill1049 Oh, it had never occurred to me. Is it because React Native will always have controlled components? At least we should add some comments regarding that.
> 
> The reason why I wanted to replace Proxy is that we lint our built file with `eslint-plugin-compat` to ensure compatibility with some old browsers and there are no good Proxy polyfill that covers all edge cases.

we do feature detect for `Proxy` in the lib (Proxy is a web only API). That's fine with old browser as well, only diff is more re-renders.

**Pull request closed without merging**